AdminAction.ts

```diff
@@ -0,0 +1,17 @@
+import { Field, Struct } from "o1js"
+
+/** Enumerates the different types of administrative actions that can be performed on a token. */
+export class AdminAction extends Struct({
+  type: Field,
+}) {
+  public static types = {
```

Comment:
Maybe extract this into a `AdminActionType` enum?

Reply:
I went the other route, and removed the type altogether. Instead, there are now separate methods to approve each individual action. That makes it easier to overwrite a particular permission (as I've done in one of the tests)